File: subsets-ii/subsets-ii.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        def back(nums,ans,temp):
            ans.append(temp)
            for i in range(len(nums)):
                if i!=0 and nums[i]==nums[i-1]:
                    continue
                back(nums[i+1:], ans, temp+[nums[i]])
        ans=[]
        back(sorted(nums),ans,[])
        return ans

File: subsets-ii/subsets-ii.cpp
class Solution {
    set<vector<int>> ans;
    void f(int cur, vector<int>& nums, vector<int> res){
        if (cur<0) {
            ans.insert(res);
            return;
        }

        f(cur-1,nums,res);
        res.push_back(nums[cur]);
        f(cur-1,nums,res);
    }
public:
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        vector<int>res;
        sort(nums.begin(),nums.end());
        f(nums.size()-1,nums,res);
        vector<vector<int>>final_res;
        for (auto it: ans){
            final_res.push_back(it);
        }        
        return final_res;
    }
};